Cover the SpiderMiddlewares.scrapy_response output change in 2.13

Before Scrapy 2.13, `SpiderMiddlewares.scrape_response()` used to return `Deferred[tuple[Response, Request, Spider]]`.

Since 2.13 (https://github.com/scrapy/scrapy/commit/1282ddf8f77299edf613679c2ee0b606e96808ce), it returns a `Deferred[Union[MutableChain[_T], MutableAsyncChain[_T]]]`.

We should cover this in the list of backward-incompatible changes in Scrapy 2.13.

I’m addressing the scrapy-zyte-api issue. For future reference, this was causing the following kind of error in scrapy-zyte-api 0.29.0:

```
crawler = <scrapy.crawler.Crawler object at 0x7effb873fe80>
request = <GET https://example.com/>, is_start_request = False

    @ensureDeferred
    async def _process_request(crawler, request, is_start_request=False):
        spider = crawler.spider
    
        await crawler.engine.scraper.open_spider(spider)
        await crawler.engine.signals.send_catch_log_deferred(
            signals.spider_opened, spider=spider
        )
    
        spider_middlewares = crawler.engine.scraper.spidermw
        if is_start_request:
            if hasattr(spider_middlewares, "process_start_requests"):
                result = await spider_middlewares.process_start_requests([request], spider)
                request = next(result)
            else:
    
                async def start():
                    yield request
    
                spider.start = start
                request = await (await spider_middlewares.process_start(spider)).__anext__()
        else:
            response = Response(request.url, request=request)
>           _, request, _ = await spider_middlewares.scrape_response(
                lambda *args: args, response, request, spider
            )
E           ValueError: not enough values to unpack (expected 3, got 2)
```

--- a/docs/news.rst
+++ b/docs/news.rst
@@ -2042,6 +2042,13 @@ Backward-incompatible changes
     ``process_start_requests()`` has been replaced by ``process_start()``.
     (:issue:`6729`)
 
+-   The ``scrape_func`` callable passed to
+    ``scrapy.core.spidermw.SpiderMiddlewareManager.scrape_response()`` is now
+    called with 2 parameters, ``response`` and ``request``, instead of 3, and
+    must return a :class:`~twisted.internet.defer.Deferred` instead of an
+    iterable.
+    (:issue:`6787`)
+
 -   The now-deprecated ``start_requests()`` method, when it returns an iterable
     instead of being defined as a generator, is now executed *after* the
     :ref:`scheduler <topics-scheduler>` instance has been created.

